pkg/utils/utils.go

```diff
@@ -632,7 +660,86 @@ func GetClusterFlavor(config *rest.Config) (constants.ClusterFlavor, error) {
 		return constants.Unknown, err
 	}
 
-	return GetCSIClusterType(clientset)
+	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
+	if !exist {
+		fmt.Printf("The VELERO_NAMESPACE environment variable is empty, assuming velero as namespace\n")
+		veleroNs = "velero"
+	}
+
+	return retrieveClusterFlavor(clientset, veleroNs)
+}
+
+// Internal method to enable unit testing.
+func retrieveClusterFlavor(clientset kubernetes.Interface, veleroNs string) (constants.ClusterFlavor, error) {
+	var err error
+	// Check for velero-vsphere-plugin-config in the velero installation namespace
+	clusterFlavor, err := GetClusterTypeFromConfig(clientset, veleroNs, constants.VeleroVSpherePluginConfig)
+	if err != nil {
+		fmt.Printf("Error received while retrieving cluster flavor from config, err: %+v\n", err)
+		clusterFlavor = constants.Unknown
+	}
+	if clusterFlavor == constants.Unknown {
+		fmt.Printf("Falling back to retrieving cluster flavor from vSphere CSI Driver Deployment\n")
+		clusterFlavor, err = GetCSIClusterType(clientset)
+		if err != nil {
+			fmt.Printf("Received error while retrieving cluster flavor from vSphere CSI Driver, err: %+v\n", err)
+			clusterFlavor = constants.Unknown
+		}
+	}
+	if clusterFlavor == constants.Unknown {
+		fmt.Printf("Failed to retrieve cluster flavor from Velero Vsphere Plugin Config or vSphere CSI Driver\n")
+		fmt.Printf("Defaulting the cluster flavor to VANILLA\n")
+		clusterFlavor = constants.VSphere
+	}
+	return clusterFlavor, nil
+}
+
+func GetClusterTypeFromConfig(kubeClient kubernetes.Interface, veleroNs string, configName string) (constants.ClusterFlavor, error) {
+	veleroPuginConfigMap, err := kubeClient.CoreV1().ConfigMaps(veleroNs).Get(context.TODO(), configName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Failed to retrieve %s configuration, err: %+v.\n", configName, err)
+		return constants.Unknown, err
+	}
+	configMap := veleroPuginConfigMap.Data
+	if flavour, ok := configMap[constants.ConfigClusterFlavor]; ok {
+		clusterFlavor := ConvertConfigClusterFlavor(flavour)
+		return clusterFlavor, nil
+	}
+	return constants.Unknown, nil
+}
+
+func ConvertConfigClusterFlavor(flavor string) constants.ClusterFlavor {
+	switch flavor {
+	case "VANILLA":
+		return constants.VSphere
```

Comment:
What is the value of constants.VSphere?

Reply:
"vSphere Kubernetes Cluster"

Reply:
Would you please help rename the constants below? It would be much more self-explanatory.
```
VSphere -> Vanilla
TkgGuest -> Guest
```

Reply:
This will change way too many files, will do it as a part of another PR. Let's keep this PR focussed just one thing.

Reply:
Sure. Would you please create a github issue or jira ticket for it? Thanks.

Reply:
Please create one and assign it to pool. Someone can pick it up when they have free cycles.